src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java

```diff
@@ -179,42 +172,52 @@ private synchronized static MongoClient getMongoClient(final String hosts)
 		return client;
 	}
 
+
+	private synchronized static MongoClient getMongoClientWithCredential(final String hosts, MongoCredential credential)
```

Comment:
Get a mongo client with a list of hosts(replicates?) and authenticate at the same time. New way of doing in in mongo3